src/main/java/com/google/sps/data/Contest.java

```diff
@@ -83,6 +93,30 @@ public static Contest fromJSONObject(DatastoreService datastore, JSONObject cont
         .build();
   }
 
+  // Convert this Contest object to a String in JSON format. Serializes its list of Candiate Entity
+  // key names into JSON objects that correspond to each Candidate's attributes.
+  public String toJsonString(DatastoreService datastore) {
+    Gson gson = new Gson();
+    JsonArray candidateJsonArray = new JsonArray();
+
+    // Since a Contest object's "candidates" collection consists only of their Datastore key names,
+    // iterate through each one to get the actual Candidate field data in JSON format.
+    Set<Long> contestCandidateIds = ImmutableSet.copyOf(this.getCandidates());
+    for (Long candidateId : contestCandidateIds) {
```

Comment:
\* no action required / FYI * 

I would consider using Java stream here, since you're really doing a series of transformations on the Set.
1) id -> key
2) key -> datastore lookup
3) entity -> json

followed by collecting all of the json strings into the json array.